hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -859,6 +869,42 @@ private void validateSnapshotsExistAndActive(final String volumeName,
     checkSnapshotActive(toSnapInfo, false);
   }
 
+  public void snapshotLimitCheck() throws IOException, OMException {
+    OmMetadataManagerImpl omMetadataManager = (OmMetadataManagerImpl) ozoneManager.getMetadataManager();
+    SnapshotChainManager snapshotChainManager = omMetadataManager.getSnapshotChainManager();
+    int currentSnapshotNum = snapshotChainManager.getGlobalSnapshotChain().size();
```

Comment:
Can't we call this inside inflightSnapshotCount.updateAndGet() function that would make this threadsafe and consistent.

Reply:
I was off the previous week so couldn't review this PR in time. Can you please make this change in a follow up jira.